packages/protocol/contracts/identity/Random.sol

```diff
@@ -43,16 +61,51 @@ contract Random is IRandom {
     }
 
     // add entropy
-    _random = keccak256(abi.encodePacked(_random, randomness));
+    addRandomness(block.number, keccak256(abi.encodePacked(_random, randomness)));
 
     commitments[proposer] = newCommitment;
   }
 
+  function addRandomness(uint bn, bytes32 randomness) internal {
```

Comment:
This is a neat trick for adapting to changes to `randomnessBlockRetentionWindow` 👍